boa/src/syntax/lexer/cursor.rs

```diff
@@ -130,6 +130,7 @@ where
     /// predicate on the ascii char
     ///
     /// The buffer is not incremented.
+    #[allow(dead_code)]
```

Comment:
Why do we allow dead code here? If it's not being used, it should be removed, right?

Same applies to line 195.